surefire-logger-api/src/main/java/org/apache/maven/plugin/surefire/log/api/Level.java

```diff
@@ -0,0 +1,83 @@
+package org.apache.maven.plugin.surefire.log.api;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Standard log levels, {@link #DEBUG} {@link #INFO} {@link #WARN} {@link #ERROR}, used in console.
+ * Test result levels {@link #FAILURE}, {@link #UNSTABLE}, {@link #SUCCESS}.
+ * Writing to console without color via {@link #NO_COLOR}.
+ *
+ * @author <a href="mailto:tibordigana@apache.org">Tibor Digana (tibor17)</a>
+ * @since 2.19.2
+ */
+public enum Level {
+    /**
+     * call {@link ColoredConsoleMessageBuilder#toColorless(CharSequence)}
+     */
+    NO_COLOR,
+
+    /**
+     * defaults to bold, cyan
+     */
+    DEBUG,
+
+    /**
+     * defaults to bold, green
+     */
+    INFO,
+
+    /**
+     * defaults to bold, yellow
+     */
+    WARN,
+
+    /**
+     * defaults to bold, red
+     */
+    ERROR,
+
+    /**
+     * defaults to bold, green
+     */
+    FAILURE,
+
+    /**
+     * defaults to bold, yellow
+     */
+    UNSTABLE,
+
+    /**
+     * defaults to bold, red
+     */
+    SUCCESS;
+
+    public static Level resolveLevel(boolean hasSuccessful, boolean hasFailure, boolean hasError,
+                                     boolean hasSkipped, boolean hasFlake) {
+        boolean isRed = hasFailure | hasError;
+        if (isRed) {
```

Comment:
you'll need to reformat this class (and NullConsoleLogger) since they don't use Maven formatting conventions then are detected by Checkstyle